### ISSUE TITLE
Issue loading sidebar content - first link always shows as empty

### ISSUE THREAD

**Issue by BanditTech** [anikethsaha/docsify-plugin#10]
#### Steps to reproduce

Load into a docsify page, click on a sidebar entry loading a file / subdirectory
refresh the page at the current location

click on another item on sidebar, 
> content within the file will not populate the sidebar as a list

click on any further sidebar items after the first
> content will populate as expected

#### What is current behaviour
When loading the page at a subdirectory, the first sidebar entry will not load its contents into the sidebar

#### What is the expected behaviour
Whether loaded into any page on the site, it will always populate the content of the file into the sidebar. It is strange that its always the first item to click.

#### Other relevant information

This issue is  occuring because of the line :
(` ${this.responseText} \n # {docsify-ignore-all} `)

I tracked down the issue to this plugin. When loading the first link on a page, it will actually load the changelog function for some reason. It will then insert the contents of the changelog into the first subsection clicked. I tried messing around with different things to try make this resolve but no clue how to make this plugin load properly. 

The way it should load is only when clicking the actual nav link.

**Comment by anikethsaha** [comment]
You meant only load the function when the button is clicked ?

**Comment by BanditTech** [comment]
yes, the plugin seems to be affecting the entire page load, somehow the event that triggers the injection of the changelog file needs to be moved to the click event on the nav button itself. It seems to now be injecting the changelog on the first click of any link.

Here is one situation to demonstrate what happens when the line to do docsify-ignore-all is not there, and it will explain more about the issue.

Load the page, click a subdirectory page

Refresh the current page with f5

click on another page within the same subdirectory

the headings from the changelog will populate inside the section you just expanded

clicking any other sections will populate their file contents as you would expect.

So its showing that the load event for the changelog is tied to any HTML onreadystatechange, instead of being tied to the button with a conditional statement.

**Comment by anikethsaha** [comment]
Ok! seems like a bug. What solution you are thinking will fix this ?

**Comment by BanditTech** [comment]
there needs to be a better way of triggering the HTML injection. As it stands, its the first redirect on the page. When using a link on the main page like /doc/page it will affect the linked page and its contents will not be loaded into sidebar.

A solution that comes to mind is to replace the behavior of the button.

On page load, fetch the contents of the changelog but do not insert them. Save the contents to a local storage for later use with the notification display.

Button is inserted into nav with the HREF launching the javascript toggling the changelog show state

put a conditional into the showstate, if its first run then insert the saved response into the contents of the div and then toggle show state

Inside the

**Comment by BanditTech** [comment]
or another sulution would be load a dummy page, use the method in place, then redirect to the actual page for the rest.

**Comment by anikethsaha** [comment]
> On page load, fetch the contents of the changelog but do not insert them. Save the contents to a local storage for later use with the notification display.

we can do this, cause we are already storing the content in localstorage. We can try to implement this and check the perf if required, then we will replace our current one with this one
